clients/admin-ui/src/features/privacy-requests/RequestTable.tsx

```diff
@@ -40,36 +65,69 @@ const useRequestTable = () => {
     dispatch(setPage(filters.page + 1));
   };
 
-  const { data, isLoading, isFetching } =
-    useGetAllPrivacyRequestsQuery(cachedFilters);
-  const { items: requests, total } = data || { items: [], total: 0 };
+  const handleCheckAll = () => {
+    const value = !checkAll;
+    dispatch(
+      setRetryRequests({
+        checkAll: value,
+        errorRequests: value ? getErrorRequests() : [],
+      })
+    );
+  };
+
+  useEffect(() => {
+    updateCachedFilters.current(filters);
+    if (isFetching && filters.status?.includes("error")) {
+      dispatch(setRetryRequests({ checkAll: false, errorRequests: [] }));
+    }
+  }, [dispatch, filters, isFetching]);
+
   return {
     ...filters,
-    isLoading,
+    checkAll,
+    errorRequests,
+    handleCheckChange,
+    handleNextPage,
+    handlePreviousPage,
+    handleCheckAll,
     isFetching,
     requests,
     total,
-    handleNextPage,
-    handlePreviousPage,
   };
 };
 
 const RequestTable: React.FC<RequestTableProps> = () => {
   const {
-    requests,
-    total,
-    page,
-    size,
+    checkAll,
+    errorRequests,
+    handleCheckChange,
     handleNextPage,
     handlePreviousPage,
+    handleCheckAll,
     isFetching,
+    page,
+    requests,
+    size,
+    total,
   } = useRequestTable();
 
   return (
     <>
       <Table size="sm">
         <Thead>
           <Tr>
+            <Th px={0}>
+              <Checkbox
+                aria-label="Select all"
+                isChecked={checkAll}
+                onChange={handleCheckAll}
+                pointerEvents={
+                  requests.findIndex((r) => r.status === "error") !== -1
```

Comment:
Good use for [some](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some)